src/main/java/net/fabricmc/loader/transformer/Stripper.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.transformer;
+
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.MethodVisitor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Strips the specified interfaces, fields and methods from a class.
+ */
+public class Stripper extends ClassVisitor {
+	private final Set<String> stripInterfaces;
```

Comment:
Can we just pass Collections here, to minimize the amount of reliance on specific interfaces?

Reply:
Even if using Collection is reducing reliance in specific interfaces, I believe Set matches the usage better than Collection. I will change it if people prefer.